### ISSUE TITLE
Inline array declaration doesn't compile

### ISSUE THREAD

**Issue by codecutout** [kirbysayshi/vash#54]
Im happy to see some razor goodness make its way to javascript, although my very first example template seems to have hit a bug

It doesn't seem to like the @[1,2,3].forEach(...). It works fine if I read the array from my model, its just when i define it inline. I would think inline array declarations would be common enough to be supported (I apologize if i'm doing something wrong here)

	<html>
		<head></head>
		<body>
			@[1,2,3].forEach(function(i){
				<p>@i number AH AH AH</p>
			})
		</body>
	</html>

**Comment by kirbysayshi** [comment]
As you've discovered, the array literal syntax isn't supported. I believe I ignored that case early on because you can do:

```js
@{
[1, 2, 3].forEach(function(i) {
  <p>@i number AH AH AH</p>
})
}
```

**Comment by kirbysayshi** [comment]
This should be implemented as of >= 0.8.0!
